File: Java/src/com/matrixmultiply/SequentialMultiply.java
package com.matrixmultiply;

public class SequentialMultiply {
  public static void multiply(int[][] matrix1, int[][] matrix2) {
    int[][] result = new int[matrix1.length][matrix2[0].length];

    for (int i = 0; i < matrix1.length; i++) {
      for (int j = 0; j < matrix2[0].length; j++) {
        result[i][j] = 0;
        for (int k = 0; k < matrix2.length; k++) {
          result[i][j] += matrix1[i][k] * matrix2[k][j];
        }
      }
    }
  }
}

File: C++/MatrixMultiplication/src/sequential_multiply.cpp
#include "sequential_multiply.hpp"

std::vector<std::vector<int>> SequentialMultiply::multiply(const std::vector<std::vector<int>> &matrix1, const std::vector<std::vector<int>> &matrix2)
{
  size_t rows = matrix1.size();
  size_t cols = matrix2[0].size();
  size_t inner = matrix2.size();
  std::vector<std::vector<int>> result(rows, std::vector<int>(cols, 0));

  for (size_t i = 0; i < rows; ++i)
  {
    for (size_t j = 0; j < cols; ++j)
    {
      for (size_t k = 0; k < inner; ++k)
      {
        result[i][j] += matrix1[i][k] * matrix2[k][j];
      }
    }
  }

  return result;
}
